src/main/java/io/codelex/Test/Exercise2/Basket.java

```diff
@@ -0,0 +1,45 @@
+package io.codelex.Test.Exercise2;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Basket<T> {
+    T items;
+
+    private List<T> basket = new ArrayList<>();
+
+    public void addToBasket(T item) {
+        if (basket.size() == 10) {
+            System.out.println("Basket is full");
+        } else if (basket.size() > 0) {
+            if (basket.get(0) == item) {
+                basket.add(item);
+            } else {
+                System.out.println("Not the same item");
+            }
+        } else {
+            basket.add(item);
+        }
+    }
+
+    public void removeFromBasket(T item) {
+        if (basket.size() == 0) {
+            System.out.println("Basket is emty");
+        } else if (basket.get(0) == item) {
```

Comment:
Šis arī ir lieks (tā pati problēma, ka viņš pārbauda vai 0 elements sakrīt ar padoto objektu.
Ko šī metode šobrīd dara, ja sarakstā ir 3 āboli (3 dažādi objekti ar vienu tipu) - `a1`, `a2` un `a3`:
Ja metodei padošu `a2`:
Viņš pārbaudīs, ka basket nav tukšs <= tas ir ok
Tad viņš pārbaudīs, vai basket.get(0) sakrīt ar padoto - un nesakritīs, jo get(0) atgriezīs `a1`, bet es padevu `a2`
Nu un faktiski viņš izvadīs - "In basket not such item", kas nav patiesība.